### ISSUE TITLE
Feature suggestion - Import compendium to folder

### ISSUE THREAD

**Issue by BadIdeasBureau** [earlSt1/vtt-macro-folders#6]
When importing a compendium full of macros, it would be great if we could import that compendium into its own folder - particularly for things like DAE SRD, which comes with a whole bunch of helper macros that just want to sit in the background and be out of the way.

**Comment by Drental** [comment]
Additional feature request on this: add the ability to import a compendium as linking macros only, so that module or system updates to helper macros don't have to be updated manually in the world.

**Comment by earlSt1** [comment]
> Additional feature request on this: add the ability to import a compendium as linking macros only, so that module or system updates to helper macros don't have to be updated manually in the world.

Ah ok, so like an "Update existing" option which replaces old macros with new ones, provided their names match?

**Comment by Drental** [comment]
not quite, I was more thinking the TAH way of... Let me grab a snippet
```
    async _executeMacroByName(name) {
        let pack = game.packs.get('pf2e.pf2e-macros');
        pack.getIndex().then(index => {
            let id = index.find(e => e.name === name)?._id;

            if (id)
                pack.getEntity(id).then(e => e.execute()
        )});
    }
```
like this, but fixed to all the right pack and name per macro

**Comment by earlSt1** [comment]
Ooooh interesting, so have an option for macros like "sync with
compendium". Thats a great idea, definitely adding that to the list of
things to do

**Comment by ravingdragoon** [comment]
I would like to add to this, that it is very confusing that the import prompt asks for a name for your import folder, but then does not create said folder and instead places the imported macros in a default folder.

**Comment by earlSt1** [comment]
> I would like to add to this, that it is very confusing that the import prompt asks for a name for your import folder, but then does not create said folder and instead places the imported macros in a default folder.

Sorry I completely forgot about this feature, was on the to-do list ages ago! I've been focusing on the integration with Compendium Folders. Adding this to the top of my feature list.

**Comment by ravingdragoon** [comment]
> > I would like to add to this, that it is very confusing that the import prompt asks for a name for your import folder, but then does not create said folder and instead places the imported macros in a default folder.
> 
> Sorry I completely forgot about this feature, was on the to-do list ages ago! I've been focusing on the integration with Compendium Folders. Adding this to the top of my feature list.

Is this still something that is being looked at, or is it something that will be taken care of in Foundry 0.9?

**Comment by earlSt1** [comment]
> > > I would like to add to this, that it is very confusing that the import prompt asks for a name for your import folder, but then does not create said folder and instead places the imported macros in a default folder.
> > 
> > 
> > Sorry I completely forgot about this feature, was on the to-do list ages ago! I've been focusing on the integration with Compendium Folders. Adding this to the top of my feature list.
> 
> Is this still something that is being looked at, or is it something that will be taken care of in Foundry 0.9?

Hi, sorry I've been struggling to stay motivated updating the module, mainly because it involves fixing stupid design decisions I made early on in the project, and fixing the 0.8.x changes towards Documents instead of Entities. I've got a development branch on my machine which fixes/enables some of the Compendium Folders integration. I'll try to get some more done tomorrow.

**Comment by earlSt1** [comment]
Hi @ravingdragoon ive just pushed an update to the test version of macro folders which enables you to use the Import All dialog on a macro compendium. Can you test by uninstalling your current version of Macro Folders and installing this version via the manifest: https://raw.githubusercontent.com/earlSt1/vtt-macro-folders/08x-update/test/module.json

You will also need to use the test version of Compendium Folders by using this manifest: https://raw.githubusercontent.com/earlSt1/vtt-compendium-folders/082-update/test/module.json

I strongly recommend backing up compendiums/worlds just in case.

**Comment by ravingdragoon** [comment]
> https://raw.githubusercontent.com/earlSt1/vtt-compendium-folders/082-update/test/module.json

@earlSt1 It seems to work fantastically! I have only tested it on a new world at this point, so I will let you know if I find any issues. Something you may want to note in release notes is that any new macros will go to the correct folder and any duplicates, I assume even if they have been updated, will go to the Default folder, rather than overwriting the existing imported macro.
